halo2-wasm/src/halo2lib.rs

```diff
@@ -329,6 +330,36 @@ impl Halo2LibWasm {
         self.to_js_assigned_values(out)
     }
 
+    pub fn to_hi_lo(&mut self, a: usize) -> Vec<u32> {
+        let a = self.get_assigned_value(a);
+        let a_val = a.value();
+        let a_bytes = a_val.to_bytes();
+
+        let mut a0_bytes = [0u8; 32];
+        let mut a1_bytes = [0u8; 32];
+        a0_bytes[..16].copy_from_slice(&a_bytes[..16]);
+        a1_bytes[..16].copy_from_slice(&a_bytes[16..]);
+        let a0 = Fr::from_bytes(&a0_bytes).unwrap();
+        let a1 = Fr::from_bytes(&a1_bytes).unwrap();
+
+        let mut builder = self.builder.borrow_mut();
+        let ctx = builder.main(0);
+
+        let a0 = ctx.load_witness(a0);
+        let a1 = ctx.load_witness(a1);
+
+        self.range.range_check(ctx, a0, 128);
+        self.range.range_check(ctx, a1, 125);
```

Comment:
If `a` is truly any element in `Fr`, I don't think this is enough. E.g., what if I give you `div_mod(r + 1, 2**128)` where `r` is the modulus of `Fr`?

You need to check that EITHER:
- `a_hi < r // 2**128` (which implies `a_hi * 2 ** 128 + a_lo < r`)
- OR (`a_hi == r // 2 ** 128` AND `a_lo < r % 2**128`)